sequencer/src/persistence/sql.rs

```diff
@@ -1343,14 +1317,478 @@ impl SequencerPersistence for Persistence {
+
+            query_builder.push_values(values.into_iter(), |mut b, (view, payload_hash, data)| {
+                b.push_bind(view).push_bind(payload_hash).push_bind(data);
+            });
+
+            offset += batch_size;
+
+            let query = query_builder.build();
+
+            let mut tx = self.db.write().await?;
+            query.execute(tx.as_mut()).await?;
+
+            tx.commit().await?;
+
+            if rows.len() < batch_size as usize {
+                break;
+            }
+        }
+
+        tracing::warn!("migrated da proposals");
+
+        let mut tx = self.db.write().await?;
+        tx.upsert(
+            "epoch_migration",
+            ["table_name", "completed"],
+            ["table_name"],
+            [("da_proposal".to_string(), true)],
+        )
+        .await?;
+        tx.commit().await?;
+
+        tracing::info!("updated epoch_migration table for da_proposal");
+
+        Ok(())
+    }
+
+    async fn migrate_vid_shares(&self) -> anyhow::Result<()> {
+        let batch_size: i64 = 1000;
+        let mut offset: i64 = 0;
+        let mut tx = self.db.read().await?;
+
+        let (is_completed,) = query_as::<(bool,)>(
+            "SELECT completed from epoch_migration WHERE table_name = 'vid_share'",
+        )
+        .fetch_one(tx.as_mut())
+        .await?;
+
+        if is_completed {
+            tracing::info!("vid_share migration already done");
+            return Ok(());
+        }
+
+        tracing::warn!("migrating vid shares..");
+        loop {
+            let mut tx = self.db.read().await?;
+            let rows =
+                query("SELECT payload_hash, data FROM vid_share ORDER BY view LIMIT $1 OFFSET $2")
+                    .bind(batch_size)
+                    .bind(offset)
+                    .fetch_all(tx.as_mut())
+                    .await?;
+
+            drop(tx);
+            if rows.is_empty() {
+                break;
+            }
+            let mut values = Vec::new();
+
+            for row in rows.iter() {
+                let data: Vec<u8> = row.try_get("data")?;
+                let payload_hash: String = row.try_get("payload_hash")?;
+
+                let vid_share: ADVZDisperseShare<SeqTypes> = bincode::deserialize(&data)?;
+                let vid_share2: VidDisperseShare<SeqTypes> = vid_share.into();
+
+                let view = vid_share2.view_number().u64() as i64;
+                let data = bincode::serialize(&vid_share2)?;
+
+                values.push((view, payload_hash, data));
+            }
+
+            let mut query_builder: sqlx::QueryBuilder<Db> =
+                sqlx::QueryBuilder::new("INSERT INTO vid_share2 (view, payload_hash, data) ");
+
+            query_builder.push_values(values.into_iter(), |mut b, (view, payload_hash, data)| {
+                b.push_bind(view).push_bind(payload_hash).push_bind(data);
+            });
+
+            offset += batch_size;
```

Comment:
I think most of the functions here still don't log any progress. I think it's fine if it's reasonably fast but if it takes more than say 10 seconds or a minute I prefer logging something about the progress.